Sources/X509/X509BaseTypes/Time.swift

```diff
@@ -47,111 +47,55 @@ enum Time: DERParseable, DERSerializable, Hashable, Sendable {
 
     @inlinable
     static func makeTime(from date: Date) throws -> Time {
-        let components = gregorianCalendar.dateComponents(in: utcTimeZone, from: date)
+        let components = date.utcDate
 
         // The rule is if the year is outside the range 1950-2049 inclusive, we should encode
         // it as a generalized time. Otherwise, use a UTCTime.
-        // These force-unwraps are safe: all the components are returned by the above call.
-        if (1950..<2050).contains(components.year!) {
-            let utcTime = try UTCTime(
-                year: components.year!,
-                month: components.month!,
-                day: components.day!,
-                hours: components.hour!,
-                minutes: components.minute!,
-                seconds: components.second!
-            )
-
+        if ((1950)..<(2050)).contains(components.year) {
+            let utcTime = try UTCTime(components)
             return .utcTime(utcTime)
         } else {
-            let generalizedTime = try GeneralizedTime(
-                year: components.year!,
-                month: components.month!,
-                day: components.day!,
-                hours: components.hour!,
-                minutes: components.minute!,
-                seconds: components.second!,
-                fractionalSeconds: 0.0
-            )
-
+            let generalizedTime = try GeneralizedTime(components)
             return .generalTime(generalizedTime)
         }
     }
 }
 
 extension Date {
     @inlinable
-    init?(_ time: Time) {
-        let maybeDate: Date?
+    init(fromUTCDate date: (year: Int, month: Int, day: Int, hours: Int, minutes: Int, seconds: Int)) {
```

Comment:
Should we introduce a struct for that type?

Reply:
I didn't want to, because we already have two very similar structs: `GeneralizedTime` and `UTCTime`.